### PR TITLE
ethclient: move TransactionOpts to avoid import internal package;

### PR DESCRIPTION
**Pull request by galaio**
### Description

This PR aims to fix the issue https://github.com/bnb-chain/bsc/issues/2722, developer cannot use `ethclient.SendTransactionConditional` directly that is caused by a `import error`.

### Changes

Notable changes: 
* ethclient: move TransactionOpts to avoid import internal package;
* ...

### CODE REVIEW

File: internal/ethapi/transaction_opts_utils.go [hunk 1]
```diff
@@ -2,52 +2,14 @@ package ethapi
...
-	BlockNumberMin *hexutil.Uint64 `json:"blockNumberMin,omitempty"`
-	BlockNumberMax *hexutil.Uint64 `json:"blockNumberMax,omitempty"`
-	TimestampMin   *hexutil.Uint64 `json:"timestampMin,omitempty"`
-	TimestampMax   *hexutil.Uint64 `json:"timestampMax,omitempty"`
-}
-
 const MaxNumberOfEntries = 1000
 
-func (o *TransactionOpts) Check(blockNumber uint64, timeStamp uint64, statedb *state.StateDB) error {
+func TxOptsCheck(o types.TransactionOpts, blockNumber uint64, timeStamp uint64, statedb *state.StateDB) error {
```
Comment (emailtovamos) [inline internal/ethapi/transaction_opts_utils.go:12]: Why these Check functions also not transferred to the `transaction_options.go`?

Comment (galaio) [reply internal/ethapi/transaction_opts_utils.go:12]: Oh, `state.StateDB` cannot transfer to low-level package `types`, so I leave these util functions here.

**Comment by buddh0** [comment]
TransactionOpts is only used by clients, 
maybe including it into dir `common` is better than `core`

**Comment by galaio** [comment]
> TransactionOpts is only used by clients, maybe including it into dir `common` is better than `core`

`common` is a very low-level package, `TransactionOpts` is related `Transaction`, so maybe `core` package is ok.

**Review by zzzckck — APPROVED** [review]

**Pull request merged**